src/Forms/BatchConvert.cs

```diff
@@ -519,202 +520,223 @@ private void buttonInputBrowse_Click(object sender, EventArgs e)
-                        isTs = true;
-                    }
                     else
                     {
-                        item.SubItems.Add(Configuration.Settings.Language.UnknownSubtitle.Title);
+                        item.SubItems.Add(format.Name);
                     }
-                }
-                else
-                {
-                    item.SubItems.Add(format.Name);
-                }
-                item.SubItems.Add("-");
+                    item.SubItems.Add("-");
 
-                if (isMkv)
-                {
-                    if (mkvCount > 0)
+                    if (isMkv)
                     {
-                        listViewInputFiles.Items.Add(item);
-                    }
+                        if (mkvCount > 0)
+                        {
+                            listViewInputFiles.Items.Add(item);
+                        }
 
-                    foreach (var lang in mkvPgs)
-                    {
-                        item = new ListViewItem(fileName);
-                        item.SubItems.Add(Utilities.FormatBytesToDisplayFileSize(fi.Length));
-                        listViewInputFiles.Items.Add(item);
-                        item.SubItems.Add("Matroska/PGS - " + lang);
-                        item.SubItems.Add("-");
-                    }
-                    foreach (var lang in mkvVobSub)
-                    {
-                        item = new ListViewItem(fileName);
-                        item.SubItems.Add(Utilities.FormatBytesToDisplayFileSize(fi.Length));
-                        listViewInputFiles.Items.Add(item);
-                        item.SubItems.Add("Matroska/VobSub - " + lang);
-                        item.SubItems.Add("-");
+                        foreach (var lang in mkvPgs)
+                        {
+                            item = new ListViewItem(fileName);
+                            item.SubItems.Add(Utilities.FormatBytesToDisplayFileSize(fi.Length));
+                            listViewInputFiles.Items.Add(item);
+                            item.SubItems.Add("Matroska/PGS - " + lang);
+                            item.SubItems.Add("-");
+                        }
+                        foreach (var lang in mkvVobSub)
+                        {
+                            item = new ListViewItem(fileName);
+                            item.SubItems.Add(Utilities.FormatBytesToDisplayFileSize(fi.Length));
+                            listViewInputFiles.Items.Add(item);
+                            item.SubItems.Add("Matroska/VobSub - " + lang);
+                            item.SubItems.Add("-");
+                        }
+                        foreach (var lang in mkvSrt)
+                        {
+                            if (comboBoxFilter.SelectedIndex == 5 && textBoxFilter.Text.Length > 0)
+                            {
+                                if (lang.Contains(textBoxFilter.Text, StringComparison.OrdinalIgnoreCase))

```

Comment:
I don't know if there is a better way to formulate this without having to write the same code twice.